### PR TITLE
pkg/rpc: remove read/write deadlines as a mechanism for request timeouts

### PR DESCRIPTION
**Pull request by jtolds**
libuplink was incorrectly setting timeouts to 10 seconds still, but should have been at least 10 minutes. the order sender was setting them to 1 hour. we don't want timeouts in uplink-side logic as it establishes a minimum rate on tcp streams.

instead of all of this, just use tcp keep alive. tcp keep alive packets are sent every 15 seconds and if the peer stops responding the connection dies. this is enabled by default with go. this will kill tcp connections when they stop working.

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

**Comment by jtolds** [comment]
kid's first PR

File: lib/uplink/uplink.go [hunk 1]
```diff
@@ -70,10 +69,6 @@ type Config struct {
 		// DialTimeout is the maximum time to wait connecting to another node.
 		// If not set, the library default (20 seconds) will be used.
 		DialTimeout time.Duration
-
-		// RequestTimeout is the maximum time to wait for a request response from another node.
-		// If not set, the library default (20 seconds) will be used.
-		RequestTimeout time.Duration
```
Comment (mniewrzal) [inline lib/uplink/uplink.go]: Maybe we should keep this parameter, without logic behind, and mark it as deprecated?

Comment (isaachess) [reply lib/uplink/uplink.go]: :+1: We at least need to be sure that supplying the option via command-line doesn't crash the program if it's removed.

Comment (jtolds) [reply lib/uplink/uplink.go]: this is the point of the volatile section though. i'm also okay with people finding out their config flag is gone this way.

**Pull request merged**